go/proxyd/server.go

```diff
@@ -83,7 +92,17 @@ func (s *Server) HandleRPC(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	backendRes, err := s.backends.Forward(ctx, req)
+  group := s.rpcMethodMappings[req.Method]
+  if group == "" {
+      // use unknown below to prevent DOS vector that fills up memory
+      // with arbitrary method names.
+      log.Info("blocked request for non-whitelisted method", "source", "ws", "method", req.Method)
+      RecordRPCError(ctx, BackendProxyd, MethodUnknown, ErrMethodNotWhitelisted)
+      writeRPCError(w, req.ID, ErrMethodNotWhitelisted)
+      return
+  }
+
+	backendRes, err := s.backendGroups[group].Forward(ctx, req)
```

Comment:
Perhaps program defensively here by first checking to make sure `group` is in `s.backendGroups` before calling `Forward`?

Reply:
You can't define an invalid method mapping in the config like that - the daemon will refuse to start.